pydantic/errors.py

```diff
@@ -438,6 +442,21 @@ class DateError(PydanticValueError):
     msg_template = 'invalid date format'
 
 
+class _DateValueError(PydanticValueError):
+    def __init__(self, *, date: date) -> None:
+        super().__init__(date=str(date))
+
+
+class DateNotInThePastError(_DateValueError):
+    code = 'date.not_in_the_past'
+    msg_template = 'date "{date}" is not in the past'
```

Comment:
As per #1421, we generally don't include the invalid value in error contexts. In v2 we will always include it.

For now in this case, to keep things simple and consistent with other errors, I think we should remove this and replace it with just `date is not in the past`.

Reply:
Do I still pass the value to the error?
```python
class _DateValueError(PydanticValueError):
    def __init__(self, *, date: date) -> None:
        super().__init__(date=str(date))
```